packages/fbjs/src/functional/partitionObjectByKey.js

```diff
@@ -18,10 +18,10 @@ var partitionObject = require('partitionObject');
  * `whitelistObjectKeys`, but eventually keeping all the keys. Returns a tuple
  * of objects `[first, second]`.
  */
-function partitionObjectByKey(
-  source: Object,
+function partitionObjectByKey<Tv>(
+  source: {[key: string]: Tv},
   whitelist: Set<string>
-): Array<Object> {
+): [{[key: string]: Tv}, {[key: string]: Tv}] {
```

Comment:
Internally we just have a `$FlowFixMe` here because upgrade codemods, but I'm curious what the right thing is. I tried to apply this change internally but some other code starts failing because we have some uses where `Tv` isn't consistent (primary use seems to be splitting a props object in a React component). So we can't take it as is. I don't know how to type that correctly though - `mixed` (instead of the type parameter) doesn't seem to work (have to add more type checks in places) and `any` is never awesome. Just changing the return signature to `[Object, Object]` to make sure the tuple is handled correctly (which seems to be the thing to fix in 0.38) and leaving the rest as it was seems to be fine. But it also seems overly generic.

cc @gabelevi to perhaps comment?